clang/lib/Sema/SemaChecking.cpp

```diff
@@ -11647,6 +11647,29 @@ static void DiagnoseFloatingImpCast(Sema &S, Expr *E, QualType T,
   }
 }
 
+static void CheckCommaOperand(Sema &S, Expr *E, QualType T, SourceLocation CC,
+                              bool Check) {
+  E = E->IgnoreParenImpCasts();
+  AnalyzeImplicitConversions(S, E, CC);
+
+  if (Check && E->getType() != T)
```

Comment:
Curious why the 2nd half of this check?  Is there some problem Checking implicit conversion where no conversion exists?

Reply:
Patterned after `CheckConditionalOperand`, but the basic idea is: if the types are the same, no need to do any extra work to check for implicit conversions.